raster/src/test/scala/geotrellis/raster/GridExtentSpec.scala

```diff
@@ -136,5 +136,28 @@ class GridExtentSpec extends FunSpec with Matchers {
         result
       }).reduce(_ && _) should be (true)
     }
+
+    it("should not crash in withResolution") {
+      def viableCellSizes(rs: RasterSource,
+                          baseCellSize: CellSize,
+                          sizes: List[CellSize],
+                          targetCellCount: Long): List[CellSize] = {
+        val width = rs.cols
+        val height = rs.rows
+
+        val isGood: CellSize => Boolean = (cs: CellSize) => {
+          val cellCount = width / (cs.width / baseCellSize.width) * (height / (cs.height / baseCellSize.height))
+          println(s"Cell count for $cs: $cellCount")
+          cellCount < targetCellCount
+        }
+
+        sizes.filter(isGood)
+      }
+      val path = baseGeoTiffPath("failing-withResolutions.tif")
+      val rsSb = GeoTiffRasterSource(path)
```

Comment:
Hm, I think you can extract `GridExtents` manually for the test case and you don't even need this test file to be used in tests at all.